unicodetools/src/test/java/org/unicode/text/UCD/TestVersionedSymbolTable.java

```diff
@@ -0,0 +1,337 @@
+    }
+
+    @Test
+    void testRegularExpressionQueries() {
+        assertThatUnicodeSet("\\p{Name=/CAPITAL LETTER/}").contains("A").doesNotContain("a");
+        assertThatUnicodeSet("\\p{Block=/^Cyrillic/}")
+                .contains("и")
+                .contains("\u1C8B")
+                .contains("\u1C8F")
+                .contains("ꙮ")
+                .doesNotContain("k");
+        assertThatUnicodeSet("\\p{scx=/Gondi/}")
+                .isEqualToUnicodeSet("[\\p{scx=Gunjala_Gondi}\\p{scx=Masaram_Gondi}]")
+                .contains("𑴀")
+                .contains("𑵠")
+                .contains("।")
+                .doesNotContain("a");
+        assertThatUnicodeSet("\\p{gc=/^P/}")
+                .isEqualToUnicodeSet("[\\p{Punctuation} \\p{Private Use} \\u2029]");
+
+        assertThatUnicodeSet("\\p{Name=/NO BREAK SPACE/}").isEmpty();
+        assertThatUnicodeSet("\\p{Name=/NO-BREAK SPACE/}")
+                .contains("\u00A0")
+                .contains("\u202F")
+                .contains("\uFEFF");
+        assertThatUnicodeSet("\\p{Script=/ Gondi/}").isEmpty();
+        assertThatUnicodeSet("\\p{Script=/_Gondi/}").contains("𑴀").contains("𑵠");
+        assertThatUnicodeSet("\\p{gc=/Cased_Letter/}").isEmpty();
+        assertThatUnicodeSet("\\p{gc=Cased_Letter}")
+                .contains("a")
+                .contains("A")
+                .doesNotContain("𒀀");
+    }
+
+    /** Helper class for testing multiple properties of the same UnicodeSet. */
+    private static class UnicodeSetTestFluent {
+        UnicodeSetTestFluent(String expression) {
+            this.expression = expression;
+            ParsePosition parsePosition = new ParsePosition(0);
+            try {
+                set = new UnicodeSet(expression);
+                set.complement().complement();
+            } catch (Exception e) {
+                exception = e;
+            }
+        }
+
+        public void isIllFormed(String messageSubstring) {
+            assertNotNull(exception, expression + " is ill-formed");
+            assertTrue(
+                    exception.getMessage().contains(messageSubstring),
+                    "Error message '"
+                            + exception.getMessage()
+                            + "' for "
+                            + expression
+                            + " contains '"
+                            + messageSubstring
+                            + "'");
+        }
+
+        public <T extends CharSequence> UnicodeSetTestFluent isEqualToUnicodeSet(
+                String expectedExpression) {
+            final var expected = new UnicodeSet(expectedExpression);
+            assertTrue(
+                    set.containsAll(expected),
+                    expected + " ⊆ " + expression + " = " + set.toPattern(true));
+            assertTrue(
+                    expected.containsAll(set),
+                    expected + " ⊇ " + expression + " = " + set.toPattern(true));
```

Comment:
optional: when these fail, it can help to show the differences, for example the non-empty set.removeAll(expected) for the first assertion